.github/scripts/pullers.py

```diff
@@ -1,41 +1,63 @@
-import re
 from typing import Optional
 
 import requests
 from github import Issue
+from requests import JSONDecodeError
 
-from utils import has_label
+from exceptions import KRSMaintenanceError
 from labels import Label
+from utils import has_label
 
 
-class OrgDataPuller:
+class KRSDataPuller:
 
     def __init__(self, krs: str):
         self.krs = krs
         self.data = self.pull_data()
 
     def pull_data(self) -> dict | None:
-        response = requests.get(f"https://api-krs.ms.gov.pl/api/krs/OdpisAktualny/{self.krs}?rejestr=S&format=json")
+        response = requests.get(
+            f"https://api-krs.ms.gov.pl/api/krs/OdpisAktualny/{self.krs}?rejestr=S&format=json"
+        )
         if response.status_code == 200:
-            return response.json()
+            try:
+                return response.json()
+            except JSONDecodeError:
+                if "Przerwa techniczna" in response.text:
+                    raise KRSMaintenanceError(
+                        f"Przerwa techniczna serwisu weryfikującego KRS. "
+                        f"Proszę [zweryfikować KRS ręcznie]"
+                        f"(https://api-krs.ms.gov.pl/api/krs/OdpisAktualny/{self.krs}?rejestr=S&format=json)."
+                    )
         else:
             raise requests.HTTPError(f"Failed to fetch data for KRS {self.krs}")
 
     @property
     def name(self):
-        return self.data.get("odpis", {}).get("dane", {}).get("dzial1", {}).get("danePodmiotu", {}).get("nazwa")
+        return (
+            self.data.get("odpis", {})
+            .get("dane", {})
+            .get("dzial1", {})
+            .get("danePodmiotu", {})
+            .get("nazwa")
+        )
 
     @staticmethod
-    def get_org_by_krs(issue: Issue, krs: str) -> Optional["OrgDataPuller"]:
+    def get_org_by_krs(issue: Issue, krs: str) -> Optional["KRSDataPuller"]:
```

Comment:
od Pythona 3.10 zalecany jest union z None :grin: + możesz wykorzystać `Self` z `typing`. Wtdy zrobiłbym z tego classmethodę i w lini 50 bezpośrednio wykorzystał `cls(krs)`.

```suggestion
    def get_org_by_krs(issue: Issue, krs: str) -> Self | None:
```

Reply:
Niestety. To by działało, ale mamy do czynienia z `staticmethod`.